.github/workflows/msbuild.yml

```diff
@@ -0,0 +1,31 @@
+name: MSBuild
```

Comment:
Can we name these based on the objective rather than the tool used?
Ex: Mono

Reply:
I really wanted to match ".NET Build" but "MSBuild Build" sounded weird. I can use whatever name you like though - what's the connection between msbuild and mono?

Reply:
When I set it up I believed `msbuild` was building with Mono but that's not necessarily the case. I was using the mono docker image so it likely was compiling with mono but it looks like the correct approach would be to compile using `csc`

https://www.mono-project.com/docs/getting-started/mono-basics/

I'd recommend renaming this workflow to `Mono Build`

We really just want to make sure anvil compiles with .NET and Mono since there are sometimes subtle differences. As we build out unit tests this will become more important.